src/components/modes/teacher/Responses.js

```diff
@@ -0,0 +1,167 @@
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Paper from '@material-ui/core/Paper';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import { Delete as DeleteIcon } from '@material-ui/icons';
+import TableCell from '@material-ui/core/TableCell';
+import TableBody from '@material-ui/core/TableBody';
+import { withTranslation } from 'react-i18next';
+import { withStyles } from '@material-ui/core/styles';
+import PropTypes from 'prop-types';
+import IconButton from '@material-ui/core/IconButton';
+import ConfirmDialog from '../../common/ConfirmDialog';
+import { deleteAppInstanceResource } from '../../../actions';
+
+class Responses extends Component {
+  static styles = theme => ({
+    root: {
+      width: '100%',
+      marginTop: theme.spacing(3),
+      overflowX: 'auto',
+    },
+    table: {
+      minWidth: 700,
+    },
+  });
+
+  state = {
+    confirmDialogOpen: false,
+  };
+
+  static propTypes = {
+    t: PropTypes.func.isRequired,
+    dispatchDeleteAppInstanceResource: PropTypes.func.isRequired,
+    userContent: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        name: PropTypes.string,
+      }),
+    ).isRequired,
+    appInstanceResources: PropTypes.arrayOf(
+      PropTypes.shape({
+        data: PropTypes.number,
+        user: PropTypes.string,
+        _id: PropTypes.number,
+      }),
+    ).isRequired,
+    classes: PropTypes.shape({
+      root: PropTypes.string,
+      table: PropTypes.string,
+    }).isRequired,
+  };
+
+  handleToggleConfirmDialog = open => () => {
+    this.setState({
+      confirmDialogOpen: open,
+    });
+  };
+
+  handleConfirmDelete = id => {
+    const { dispatchDeleteAppInstanceResource } = this.props;
+    dispatchDeleteAppInstanceResource(id);
+    this.handleToggleConfirmDialog(false)();
+  };
+
+  milliToHMS = duration => {
```

Comment:
I would rename this `formatTime` and describe in a comment:

`// takes millis and outputs hh:mm:ss`